### DISCUSSION TITLE
Does/should SC change behaviour according to Android "Data Saver" setting?

### DISCUSSION THREAD

**Discussion by timothywashere** [streetcomplete/StreetComplete#3556] [General]
I don't believe that SC currently changes it's behaviour according to the "Data Saver" setting. Correct me if I'm wrong. ( I couldn't find any discussions on GitHub either. )

Maybe I am the oddity and everyone else has unlimited mobile data, but I'm still on PAYG and so restrict my mobile data usage as much as possible. 

I know some people want SC to download as much as possible (#1829) and when I'm on WiFi I do too. 

I have two tactics that I experiment with:
I turn mobile data off and on quickly to download new SC quests or background tiles. (This has become more of a pain in Android 12). Downside is that other apps start downloading stuff in the background too.
Other tactic is to leave mobile data turned on and turn on the android feature "Data Saver" (maybe called different things on different brand phones). SC then downloads data every few minutes, probably more often than if I do it manually, but less hassle.

Some questions or discussion points...
Does/should the amount of data SC downloads automatically change with mobile/wifi, metered/unmetered and/or data-saver setting? 

Does/should the amount of data SC downloads via "Scan for quests here" change with mobile/wifi, metered/unmetered and/or data-sever setting?

I think the data-saver setting is primarily for background data usage, so you might not feel it applies to SC, but I don't see any reason why the foreground behaviour can't be tweaked too if appropriate, given that you know the user is trying to save data.

**Comment by mnalis** [comment]
Perhaps when data saver is enabled, automatic quest downloading can be disabled, instead relying on user to manually initiate `Scan for quests here` (it might even do that for automatic map downloading, but that might be more complex.)

My only worry is how to communicate that to the users, as SC target audience is usually not technically savvy, so they might be very confused why _"SC doesn't work anymore"_

**Comment by Lee-Carre** [comment]
Besides downloading less data, an obvious one would be to set the configuration-variable for when to upload, to only-on-802.11 .

**Comment by HolgerJeromin** [comment]
The quest download area / radius could/should be reduced

**Reply by smichel17** [reply]
Not necessarily… multiple small downloads may overlap more and download the same area twice. I'd say it needs testing.

**Comment by matkoniecz** [comment]
I am worried that this would result in people confused why SC is behaving in a substandard way and not connecting it to some system setting - there is already repeated confusion with sound setting following system setting.

**Comment by Helium314** [comment]
To actually answer the questions:

> Does/should the amount of data SC downloads automatically change with mobile/wifi, metered/unmetered and/or data-saver setting?

On mobile data usually less data is downloaded, see https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/data/download/AVariableRadiusStrategy.kt for details.
`maxDownloadAreaInKm2` is 10 for mobile data, 20 for WiFi
`desiredScoredMapDataCountInVicinity` is 7500 for mobile data, 15000 for WiFi

Values could be further reduced if data saver is enabled, and should not be perceived as bug.
But maybe @smichel17 is right and you might end up with more data downloaded distributed over multiple downloads. Map tiles should not be downloaded again, but OSM data might be.

> Does/should the amount of data SC downloads via "Scan for quests here" change with mobile/wifi, metered/unmetered and/or data-sever setting?

This downloads the full map tiles contained in the visible area (means: usually a bit more than you see on screen). I think this should not be changed.

**Comment by timothywashere** [comment]
My experiments so far show very little usage from the local scans, when I have prescanned the area already on WiFi
Still will probably have to keep mobile data off because other apps are still using data when they shouldn't 
This is from a couple hours in an urban area
![Screenshot_20211210-103645](https://user-images.githubusercontent.com/87654968/145560891-d0f60003-f3d5-47d0-a1d5-3fc9c94d6729.png)

09 Dec 2021 	35.0664MB 	£0.367654968
